ui/src/app/applications/components/pod-logs-viewer/pod-logs-viewer.tsx

```diff
@@ -13,18 +17,207 @@ export const PodsLogsViewer = (props: {applicationName: string; pod: models.Reso
+                            {copy === '' && (
+                                <React.Fragment>
+                                    COPY <i className='fa fa-clipboard' />
+                                </React.Fragment>
+                            )}
+                        </div>
+                        <div
+                            className={`argo-button argo-button--base${prefs.appDetails.followLogs ? '' : '-o'}`}
+                            onClick={() => {
+                                const follow = !prefs.appDetails.followLogs;
+                                services.viewPreferences.updatePreferences({...prefs, appDetails: {...prefs.appDetails, followLogs: follow}});
+                                if (follow) {
+                                    setPage(0);
+                                }
+                                loader.reload();
+                            }}>
+                            FOLLOW {prefs.appDetails.followLogs && <i className='fa fa-check' />}
+                        </div>
+                        <div
+                            className='argo-button argo-button--base-o'
+                            onClick={() => {
+                                const inverted = prefs.appDetails.darkMode;
+                                services.viewPreferences.updatePreferences({...prefs, appDetails: {...prefs.appDetails, darkMode: !inverted}});
+                            }}>
+                            {prefs.appDetails.darkMode ? <i className='fa fa-sun' /> : <i className='fa fa-moon' />}
+                        </div>
+                    </div>
+                    <DataLoader
+                        ref={l => (loader = l)}
+                        loadingRenderer={() => (
+                            <div className={`pod-logs ${prefs.appDetails.darkMode ? 'pod-logs--inverted' : ''}`}>
+                                {logNavigators({}, prefs.appDetails.darkMode, null)}
+                                <pre style={{height: '95%', textAlign: 'center'}}>Loading...</pre>
+                            </div>
+                        )}
+                        load={() => {
+                            setLines([]);
+                            return services.applications.getContainerLogs(
+                                props.applicationName,
+                                props.pod.metadata.namespace,
+                                props.pod.metadata.name,
+                                container.name,
+                                maxLines * (page + 1),
+                                prefs.appDetails.followLogs,
+                                maxLines
+                            );
+                        }}>
+                        {log => {
+                            if (isRunning && !(!prefs.appDetails.followLogs && lines.length >= maxLines)) {
+                                const tmp = lines;
+                                tmp.push(log.content);
```

Comment:
We need to remove lines from the top of the page if number of lines exceeds the max page size. Otherwise page keeps growing indefinitely. 

![image](https://user-images.githubusercontent.com/426437/105133856-4f812e80-5aa2-11eb-914c-a16689d46947.png)